apis/v1beta1/workspace_types.go

```diff
@@ -107,6 +107,10 @@ type WorkspaceParameters struct {
 	// +optional
 	Entrypoint string `json:"entrypoint"`
 
+	// Variable to show a plan on CR based on condition
+	// +kubebuilder:default=false
+	ShowPlan bool `json:"showPlan"`
```

Comment:
This field should be optional, or introducing it will be a breaking change. Optional fields should be pointers with the `omitempty` tag, and the `// +optional` comment.

```suggestion
	// +optional
	ShowPlan *bool `json:"showPlan,omitempty"`
```

Also, is `showPlan` the right name for this field? I would consider `includePlan`. Perhaps subjective, but "Show" feels more like something a CLI would do.

Finally, how confident are you that this field could only ever have two possible states (i.e. show plan or don't show plan)? For example is it possible we'd ever want to include the plan encoded using a different format?


Quoting the [Kubernetes API conventions document](https://github.com/kubernetes/community/blob/4aac5c07b/contributors/devel/sig-architecture/api-conventions.md#primitive-types):

> Think twice about bool fields. Many ideas start as boolean but eventually trend towards a small set of mutually exclusive options. Plan for future expansions by describing the policy options explicitly as a string type alias

Reply:
we cannot make it optional since we need to show plan based on this variable. Also by default bool variable is assigned to false

Reply:
I agree that in future there will be more encoding options but still we will be sticking to the conditions. Anyways, if we introduced opinionated encoding options which needs input from the user so we want to introduce new variable

Reply:
> we cannot make it optional since we need to show plan based on this variable.

You can. Treat the field being unset as false.

> I agree that in future there will be more encoding options but still we will be sticking to the condition to show or not. Anyways, if we introduced opinionated encoding options which needs input from the user so we will be introducing new variable

If there will be more encoding options, it would be better to use an enum field than several boolean fields.

Reply:
Then for conditionally showing the plan, we will be checked if the field is set or not. right ?

Reply:
https://pkg.go.dev/k8s.io/utils/ptr#Deref

You can use something like the above:

```go
show := ptr.Deref[bool](mr.Spec.ShowPlan, false)
```

This will return false if the pointer is nil (i.e. unset) or points to a false value. It will return true if the pointer points to a true value.

Reply:
Done @negz